packages/clarity-tutorials/contracts/rocket-tutorial/rocket-market.scm

```diff
@@ -0,0 +1,138 @@
+
+;;;; Rocket-Market
+
+;;; Storage
+(define-map rockets-info
+  ((rocket-id int)) 
+  ((owner principal)))
+(define-map rockets-count
+  ((owner principal))
+  ((count int)))
+(define-map factory-address
+  ((id int))
+  ((address principal)))
+
+;;; Internals
+
+;; Gets the amount of rockets owned by the specified address
+;; args:
+;; @account (principal) the principal of the user
+;; returns: int
+(define (balance-of (account principal))
+  (let ((balance
+      (get count 
+        (fetch-entry rockets-count (tuple (owner account))))))
+    (if (eq? balance 'null) 0 balance)))
+
+;; Check if the transaction has been sent by the factory-address
+;; returns: boolean
+(define (is-tx-from-factory)
+  (let ((address
+    (get address 
+        (fetch-entry factory-address (tuple (id 0))))))
+    (eq? tx-sender address)))
+
+;; Gets the owner of the specified rocket ID
+;; args:
+;; @rocket-id (int) the id of the rocket to identify
+;; returns: principal
+(define (owner-of (rocket-id int)) 
+  (get owner 
+    (fetch-entry rockets-info (tuple (rocket-id rocket-id)))))
+
+;; Register a rocket
+;; args:
+;; @new-owner (principal) the principal of the new owner
+;; @rocket-id (int) the id of the rocket to transfer
+;; returns: boolean
+(define (register-rocket! (new-owner principal) (rocket-id int))
+  (let ((current-balance (balance-of new-owner)))
+    (begin
+      (set-entry! rockets-info 
+        (tuple (rocket-id rocket-id))
+        (tuple (owner new-owner))) 
+      (set-entry! rockets-count 
+        (tuple (owner new-owner))
+        (tuple (count (+ 1 current-balance)))) 
+      'true)))
+
+;; Release a rocket
+;; args:
+;; @owner (principal) the principal of the current owner
+;; @rocket-id (int) the id of the rocket to release
+;; returns: boolean
+(define (release-rocket! (owner principal) (rocket-id int))
+  (let ((current-balance (balance-of owner)))
+    (begin
+      (set-entry! rockets-count 
+        (tuple (owner owner))
+        (tuple (count (- current-balance 1)))) 
+      'true)))
+
+;;; Public functions
+
+;; Transfers rocket to a specified principal
+;; Once owned, users can trade their rockets on any unregulated black market
+;; args:
+;; @recipient (principal) the principal of the new owner of the rocket
+;; @rocket-id (int) the id of the rocket to trade
+;; returns: boolean
+(define-public (transfer (recipient principal) (rocket-id int))
+  (if (and 
+        (eq? (owner-of rocket-id) tx-sender)
+        (not (eq? recipient tx-sender)))
+    (and
+      (release-rocket! tx-sender rocket-id)
+      (register-rocket! recipient rocket-id))
+    'false))
+
+;; Mint new rockets
+;; This function can only be called by the factory.
+;; args:
+;; @owner (principal) the principal of the owner of the new rocket
+;; @rocket-id (int) the id of the rocket to mint
+;; @size (int) the size of the rocket to mint
+;; returns: boolean
+(define-public (mint! (owner principal) (rocket-id int) (size int))
+  (and
```

Comment:
Please use `(if` instead of `(and` here.

Reply:
Feedback addressed with https://github.com/blockstack/clarity-js-sdk/pull/4/commits/ac1ddc208ed46c2339fecdacb11cea8d86b66030